apps/web/src/design-system/table/Table.tsx

```diff
@@ -91,8 +112,48 @@ export function Table<T extends object>({
             pageCount: Math.ceil(total / pageSize),
           }
         : {}),
+      ...(withSelection && {
+        autoResetSelectedRows: false,
+        initialState: { selectedRowIds: initialSelectedRows ?? {} },
+      }),
     } as any,
-    usePagination
+    usePagination,
+    useRowSelect,
```

Comment:
this hook allows us to implement the selection functionality... in the future we could implement the radio buttons selection if we would like